tensorboard/plugins/pr_curve/pr_curve_demo.py

```diff
@@ -0,0 +1,198 @@
+
+  # Assign each color a vector of 3 booleans based on its true label.
+  labels = tf.concat([
+      tf.tile(tf.constant([[True, False, False]]), (number_of_reds, 1)),
+      tf.tile(tf.constant([[False, True, False]]), (number_of_greens, 1)),
+      tf.tile(tf.constant([[False, False, True]]), (number_of_blues, 1))
+  ], axis=0)
+
+  # We introduce 3 normal distributions. They are used to predict whether a
+  # color falls under a certain class (based on distances from corners of the
+  # color triangle). The distributions vary per color. We have the distributions
+  # narrow over time.
+  iteration = tf.Variable(0.0)
+  increment_iteration = tf.assign_add(iteration, 1.0)
+  with tf.control_dependencies([increment_iteration]):
+    red_predictor = tf.distributions.Normal(
+        loc=0., scale=tf.constant(40.) - iteration)
+    green_predictor = tf.distributions.Normal(
+        loc=0., scale=tf.constant(60.) - iteration)
+    blue_predictor = tf.distributions.Normal(
+        loc=0., scale=tf.constant(100.) - iteration)
+
+  # Make predictions (assign 3 probabilities to each color based on each color's
+  # distance to each of the 3 corners).
+  examples = tf.concat([true_reds, true_greens, true_blues], axis=0)
+  probabilities_colors_are_red = red_predictor.cdf(
+      tf.norm(examples - tf.constant([255., 0, 0]), axis=1))
+  probabilities_colors_are_green = green_predictor.cdf(
+      tf.norm(examples - tf.constant([0, 255., 0]), axis=1))
+  probabilities_colors_are_blue = blue_predictor.cdf(
+      tf.norm(examples - tf.constant([0, 0, 255.]), axis=1))
+  predictions = tf.concat([
+      tf.expand_dims(probabilities_colors_are_red, 1),
+      tf.expand_dims(probabilities_colors_are_green, 1),
+      tf.expand_dims(probabilities_colors_are_blue, 1)
+    ], axis=1)
+
+  if doubly_weigh_blues:
+    weights = tf.concat([
+      tf.ones((number_of_reds, 1)),
+      tf.ones((number_of_greens, 1)),
+      2 * tf.ones((number_of_blues, 1)),
+    ], axis=0)
+  else:
+    weights = None
+
+  # This is the crucial piece. We write data required for generating PR curves.
+  summary.op(
+      tag='colors',
+      labels=labels,
+      predictions=predictions,
+      num_thresholds=50,
+      num_classes=3,
+      weights=weights)
+  merged_summary_op = tf.summary.merge_all()
+  events_directory = os.path.join(logdir, run_name)
+  sess = tf.Session()
+  writer = tf.summary.FileWriter(events_directory, sess.graph)
+
+  # We run 10 steps.
+  sess.run(tf.global_variables_initializer())
+  for step in xrange(steps):
+    writer.add_summary(sess.run(merged_summary_op), step)
+
+  writer.close()
+
+
+def run_all(logdir, steps, verbose=False):
+  """Generate PR curve summaries.
+
+  Arguments:
+    logdir: The directory into which to store all the runs' data.
+    steps: The number of steps to run for.
+  """
+  # First, we generate data for a PR curve that assigns even weights for
+  # predictions of all classes.
+  run_name = 'evenly_weighed'
```

Comment:
nit: It's more standard to say ["evenly weighted" than "evenly weighed"][1] (and also ["weighted" than "weighed" alone][2], but less overwhelmingly).


[1]: https://books.google.com/ngrams/graph?content=evenly+weighed%2Cevenly+weighted&year_start=1800&year_end=2000&corpus=15&smoothing=3&share=&direct_url=t1%3B%2Cevenly%20weighed%3B%2Cc0%3B.t1%3B%2Cevenly%20weighted%3B%2Cc0
[2]: https://books.google.com/ngrams/graph?content=weighed%2Cweighted&year_start=1800&year_end=2000&corpus=15&smoothing=3&share=&direct_url=t1%3B%2Cweighed%3B%2Cc0%3B.t1%3B%2Cweighted%3B%2Cc0